latch_config/config.py

```diff
@@ -71,13 +165,19 @@ def read_config(x: type[T], env_prefix: str = "") -> T:
                 if parser is not None:
                     val = parser(env_val)
                 else:
-                    val = typ(env_val)
+                    val = parse_config_val(typ, env_val)
 
-        if val is None:
+        if val is NoValue:
             if f.default != MISSING:
                 val = f.default
             else:
-                raise RuntimeError(f"missing value for '{f.name}' (${env_name})")
+                try:
+                    # check if typ is optional and allow missing the env var in that case
+                    val = parse_config_val(typ, None)
+                except NestyError as e:
+                    raise RuntimeError(
+                        f"missing value for '{f.name}' (${env_name})"
+                    ) from e
```

Comment:
```suggestion
                    )
```
no point sending the parsing error up imo